### PR TITLE
feat: CI: capture core dumps

### PR DESCRIPTION
**Pull request by gebner**
Debugging help for #1914.

### CODE REVIEW

**Comment by Kha** [comment]
Very cool.

>  Notably, the artifact only contains the lean binary and libleanshared.so. This doesn't work for the compiled tests for example.

Couldn't we copy every `$progbin` to `coredumps/`?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -233,6 +240,22 @@ jobs:
         if: matrix.name == 'Linux'
       - name: CCache stats
         run: ccache -s
+      - name: Show stacktrace for coredumps
+        if: ${{ failure() }} && matrix.os == 'ubuntu-latest'
+        run: |
+          for c in coredumps/*; do
+            progbin="$(file $c | sed "s/.*execfn: '\([^']*\)'.*/\1/")"
+            echo bt | $GDB/bin/gdb $progbin $c || true
```
Comment (Kha) [inline .github/workflows/ci.yml:248]:
```suggestion
            echo bt | $GDB/bin/gdb -q $progbin $c || true
```

**Review by Kha — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gebner) [reply .github/workflows/ci.yml:248]: Surprisingly, not.  (At least I got a "command not found" error before.)

**Comment by gebner** [comment]
> Couldn't we copy every `$progbin` to `coredumps/`?

This line is a lot more fragile than it looks like.  I had hoped it would work better but for the compiled tests, I get a relative path here and it's not clear what to copy.  Weirdly, there is no officially supported way to get the executable path from a core dump.

**Comment by gebner** [comment]
I'm going to put the PR on the queue now so that we can debug the segfault.  Additional improvements are welcome!

**Pull request merged**